src/Serializer/ItemNormalizer.php

```diff
@@ -30,7 +30,7 @@
     public function denormalize($data, $class, $format = null, array $context = [])
     {
         // Avoid issues with proxies if we populated the object
-        if (isset($data['id']) && !isset($context['object_to_populate'])) {
+        if (isset($data['id']) && !isset($context[static::OBJECT_TO_POPULATE])) {
```

Comment:
You need to define this constant then? I'd use `self` and not `static` as this class is `final`. Also, it might be worth updating the other usages as well.

Reply:
Thank you for suggesting that. I know that OBJECT_TO_POPULATE already define in ```Symfony\Component\Serializer\Normalizer\AbstractNormalizer```. Also static better than self. And Yes, I will edit other place as well.

Reply:
@sroze. Is there anything wrong? Please let me know. Thanks

Reply:
Oh good point, it's in Symfony's class, sorry! 👍 I'm still in favour of `self` as `static` is to allow child classes.

Reply:
Please review this class: ```Symfony\Component\Serializer\NormalizerAbstractNormalizer```. Symfony use static event in the same class.

Reply:
They use `static` because the class is abstract (extensible) but ours is final (not extensible) so you should use `self` instead.

Reply:
Ok understood, thanks for explanation.